test/topology_experimental_raft/test_raft_service_levels.py

```diff
@@ -0,0 +1,66 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+import pytest
+import time
+import asyncio
+import logging
+from test.pylib.util import unique_name, wait_for_cql_and_get_hosts
+from test.pylib.manager_client import ManagerClient
+from test.pylib.internal_types import ServerInfo
+from cassandra import ConsistencyLevel
+from cassandra.query import SimpleStatement
+
+
+logger = logging.getLogger(__name__)
+
+async def trigger_snapshot(manager: ManagerClient, servers: [ServerInfo]) -> None:
+    for server in servers:
+        cql = manager.get_cql()
+        host = cql.cluster.metadata.get_host(server.ip_addr)
+        group0_id = (await cql.run_async(
+            "select value from system.scylla_local where key = 'raft_group0_id'",
+            host=host))[0].value
+        await manager.api.client.post(f"/raft/trigger_snapshot/{group0_id}", host=server.ip_addr)
+
+@pytest.mark.asyncio
+async def test_service_levels(manager: ManagerClient):
+    """
+        Cluster with 3 nodes.
+        Add 10 service levels. Start new server and it should get a snapshot on bootstrap.
+        Stop 3 `old` servers and query the new server to validete if it has the same service levels.
+    """
+    servers = await manager.servers_add(3)
+    cql = manager.get_cql()
+    await wait_for_cql_and_get_hosts(cql, servers, time.time() + 60)
+    await manager.servers_see_each_other(servers)
+
+    sls = ["sl" + unique_name() for _ in range(10)]
+    for sl in sls:
+        await cql.run_async(f"CREATE SERVICE LEVEL {sl}")
+
+    # drop non-existing service levels with CL ALL only to sync up the nodes    
+    stmt = SimpleStatement("DROP SERVICE LEVEL IF EXISTS no_such_sl", consistency_level=ConsistencyLevel.ALL)
+    await cql.run_async(stmt)
```

Comment:
I don't understand, how does it sync up all nodes? It will only cause a read barrier on the node that executes this statement, and won't on others.

If you want all nodes to observe the most recent data, you should issue a `read_barrier` _on each node_. We have a helper function for issuing a read barrier on one node in `test/pylib/utils.py`.

Reply:
actually it's not needed at all